packages/api/internal/template-manager/template_manager.go

```diff
@@ -125,39 +126,64 @@ func (tm *TemplateManager) BuildsStatusPeriodicalSync(ctx context.Context) {
 	}
 }
 
-func (tm *TemplateManager) getBuilderClient(clusterID *uuid.UUID, nodeID *string, placement bool) (*grpclient.GRPCClient, metadata.MD, PlacementLogsProvider, error) {
+func (tm *TemplateManager) GetBuildClient(clusterID *uuid.UUID, nodeID *string, placement bool) (*BuildClient, error) {
 	if clusterID == nil || nodeID == nil {
-		// build placement requires healthy template builder
-		if placement && tm.GetLocalClientStatus() != infogrpc.ServiceInfoStatus_OrchestratorHealthy {
-			zap.L().Error("Local template manager is not fully healthy, cannot use it for placement new builds")
-			return nil, nil, nil, ErrLocalTemplateManagerNotAvailable
-		}
+		return tm.GetLocalBuildClient(clusterID, nodeID, placement)
+	} else {
+		return tm.GetClusterBuildClient(*clusterID, *nodeID)
+	}
+}
 
-		// for getting build information only not valid state is getting already unhealthy builder
-		if tm.GetLocalClientStatus() == infogrpc.ServiceInfoStatus_OrchestratorUnhealthy {
-			zap.L().Error("Local template manager is unhealthy")
-			return nil, nil, nil, ErrLocalTemplateManagerNotAvailable
-		}
+func (tm *TemplateManager) GetLocalBuildClient(clusterID *uuid.UUID, nodeID *string, placement bool) (*BuildClient, error) {
```

Comment:
this doesn't need cluster nor node id

Reply:
I think the placement logic doesn't make sense here, the logic is different for BYOC and it should be probably handled inside the build function (you have to handle this there anyway as the status can change in the meantime)